virttest/libvirt_xml/devices/hostdev.py

```diff
@@ -42,14 +50,26 @@ def new_source(self, **dargs):
             new_one.untyped_address = new_address
         if self.hostdev_type == 'scsi':
             new_one.adapter_name = dargs.pop("adapter_name", None)
+            new_one.protocol = dargs.pop("protocol", None)
+            new_one.source_name = dargs.pop("source_name", None)
+            new_one.host_name = dargs.pop("host_name", None)
+            new_one.host_port = dargs.pop("host_port", None)
+            auth_args = {'auth_user': dargs.pop('auth_user', None),
+                         'secret_type': dargs.pop('secret_type', None),
+                         'secret_uuid': dargs.pop('secret_uuid', None),
+                         'secret_usage': dargs.pop('secret_usage', None)
+                         }
+            new_auth = new_one.new_auth(**auth_args)
+            new_one.auth = new_auth
             new_address = new_one.new_untyped_address(**dargs)
```

Comment:
Should be changed to:
```
if dargs:                                                           
    new_address = new_one.new_untyped_address(**dargs)              
    new_one.untyped_address = new_address  
```
since for following xml, we do not need a address object:
`  <hostdev mode='subsystem' type='scsi'>
    <source protocol='iscsi' name='iqn.2014-08.com.example:iscsi-nopool/1'>
      <host name='example.com' port='3260'/>
      <auth username='myuser'>
        <secret type='iscsi' usage='libvirtiscsi'/>
      </auth>
    </source>
    <address type='drive' controller='0' bus='0' target='0' unit='0'/>
  </hostdev>`